packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -261,15 +260,16 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
     }
 
     private addTags = (value: string) => {
-        const { onAdd, onChange, values } = this.props;
+        const { inputValue, onAdd, onChange, values } = this.props;
         const newValues = this.getValues(value);
-        let shouldClearInput = Utils.safeInvoke(onAdd, newValues);
+        let shouldClearInput = inputValue === undefined;
+        shouldClearInput = Utils.safeInvoke(onAdd, newValues) !== false && shouldClearInput;
```

Comment:
collapse these two to one line?
`let shouldClearInput = Utils.safeInvoke(onAdd, newValues) !== false && inputValue === undefined;`